tool/build.dart

```diff
@@ -176,15 +201,21 @@ _main(List<String> argv) async {
 
     await pushd("tmp/${rname}");
 
+    var branch = automated["branch"];
+
+    if (branch == null) {
+      branch = "HEAD";
+    }
+
     // Pre-Check for References
-      {
+    {
       String rev;
       var out = await exec(
         "git",
         args: [
           "ls-remote",
           repo,
-          "HEAD"
+          branch
```

Comment:
`git ls-remote HEAD` and `git ls-remote some-branch` each work as expected, returning the last commit on the remote repository. This is used to quickly bypass up-to-date builds.